### ISSUE TITLE
what is the best threshold for face search

### ISSUE THREAD

**Issue by wuxuedaifu** [mk-minchul/AdaFace#161]
for AdaFace, what is the best smilarity threshold for face search?

**Comment by mk-minchul** [comment]
It depends on the evaluation setting. But for one example, checking out the threshold that gives the best verification accuracy over 5 fold cross validations of popular verificaiton dataset (LFW, CFPFP, etc) should give you good number to work with. In my experience, 0.3 was a good number, but it is a rough guess.

**Comment by macqueen09** [comment]
After extracting features during model reasoning, what are the recommended methods to process the output features for comparing the similarity of faces?

```
@torch.no_grad()
def extract_feature(model, batch):
    batch = batch.to(device)
    # batch = torch.stack(processed_imgs)  # (batch_size, C, H, W)
    feature, _  = model(batch)
    features = feature.cpu().numpy()
    normalized_features = features / np.linalg.norm(features, axis=1, keepdims=True)
    return normalized_features
```
after that i use python 
 import faiss
to search face in a bank:
```
res = faiss.StandardGpuResources()
index= faiss.read_index(INDEX_FILE_FACE)

def search_face_in_index(feature, index, filenames):
    feature = feature.reshape(1, -1).astype("float32")
    distances, indices = index.search(feature, 1)
    best_match_idx = indices[0][0]
    best_match_file = filenames[best_match_idx]
    best_distance = distances[0][0]
    return best_match_file, best_distance
```
@mk-minchul
